### PR TITLE
fix: Send ppom metrics when transaction is cancelled.

### LINKED ISSUE #8824: [Bug]: PPOM - No blockaid data is included in Metrics events for transactions triggered from inside the wallet
### Describe the bug

**Problem:** there is no Blockaid information on metric events for transactions triggered from inside the wallet

### Expected behavior

We should have the same metrics as we have for transactions triggered from a dapp, for the following events:
- Tx added
- Tx rejected
- Tx acceptdd

### Screenshots/Recordings

https://github.com/MetaMask/metamask-mobile/assets/54408225/12153f52-baeb-4344-b6f6-0c00c2a1d1fc

### Steps to reproduce

1. Enable Blockiad
2. Enable metrics
3. Send a malicious tx from inside the wallet
4. See no blockaid info is included in the metrics event

### Error messages or log output

_No response_

### Version

7.17.0

### Build type

None

### Device

Pixel 6 

### Operating system

Android

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by segun**
## **Description**

We should send PPOM/Blockaid metrics parameters when a transaction from wallet is rejected. Right now we only send the metrics when the transaction is approved.

## **Related issues**

Fixes: [#8824](https://github.com/MetaMask/metamask-mobile/issues/8824)

## **Manual testing steps**

1. Enable Blockiad
2. Enable metrics
3. Initiate a malicious tx from inside the wallet
4. Cancel/Reject the transaction
5. See that PPOM Metrics is sent in the logs

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/54408225/12153f52-baeb-4344-b6f6-0c00c2a1d1fc

### **After**

https://github.com/MetaMask/metamask-mobile/assets/44811/efea6aa9-96f7-4b57-915c-5fe946daabdf

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/util/blockaid/index.ts [hunk 1]
```diff
@@ -67,3 +67,17 @@ export const getBlockaidMetricsParams = (
 
   return additionalParams;
 };
+
+export const withBlockaidMetricsParams = (transaction: any) => {
```
Comment (pedronfigueiredo) [inline app/util/blockaid/index.ts:71]: Do we have the Transaction type we can use here, instead of `any`?

Comment (pedronfigueiredo) [inline app/util/blockaid/index.ts:71]: [nit] rename function to `getBlockaidMetricsParams`

Comment (segun) [reply app/util/blockaid/index.ts:71]: There's already a `getBlockaidMetricsParams` defined in that utils class, I renamed to `getBlockaidTransactionMetricsParams`

Comment (segun) [reply app/util/blockaid/index.ts:71]: There's a`TransactionMeta` type in transaction controller, but it doesn't include `currentTransactionSecurityAlertResponse`.

Maybe I can define a type that extends it.

File: app/components/UI/Navbar/index.js [hunk 2]
```diff
@@ -527,9 +529,13 @@ export function getSendFlowTitle(
   });
   const rightAction = () => {
     const providerType = route?.params?.providerType ?? '';
+    const additionalTransactionMetricsParams = transaction
+      ? withBlockaidMetricsParams(transaction)
+      : {};
```
Comment (jpuri) [inline app/components/UI/Navbar/index.js:534]: `withBlockaidMetricsParams` return `{}` by default, thus above can be simplified.

Comment (segun) [reply app/components/UI/Navbar/index.js:534]: oh yeah, you're right

File: app/util/blockaid/index.ts (hunk 1 above)
Comment (jpuri) [reply app/util/blockaid/index.ts:71]: I agree `withBlockaidMetricsParams` name is not very intuitive, may be `getCurrentTransactionBlockaidMetricsParams`

**Review by pedronfigueiredo — APPROVED** [review]

**Review by jpuri — APPROVED** [review]

**Pull request merged**